sdk/core/azure-core-test/src/main/java/com/azure/core/test/InterceptorManager.java

```diff
@@ -60,44 +65,100 @@
+        } else if (allowedToRecordValues) {
+            this.recordedData = new RecordedData();
+        } else {
+            this.recordedData = null;
+        }
+    }
 
-        this.recordedData = testMode == TestMode.PLAYBACK
-            ? readDataFromFile()
-            : new RecordedData();
+    /**
+     * Creates a new InterceptorManager that replays test session records. It takes a set of
+     * {@code textReplacementRules}, that can be used by {@link PlaybackClient} to replace values in a {@link
+     * NetworkCallRecord#getResponse()}.
+     *
+     * The test session records are read from: "<i>session-records/{@code testName}.json</i>"
+     *
+     * @param testName Name of the test session record.
+     * @param textReplacementRules A set of rules to replace text in {@link NetworkCallRecord#getResponse()} when
+     * playing back network calls.
+     * @throws UncheckedIOException An existing test session record could not be located or the data could not be
+     * deserialized into an instance of {@link RecordedData}.
+     * @throws NullPointerException If {@code testName} or {@code textReplacementRules} is {@code null}.
+     * @deprecated Use {@link #InterceptorManager(String, Map, boolean)} instead.
+     */
+    @Deprecated
+    public InterceptorManager(String testName, Map<String, String> textReplacementRules) {
+        this(testName, textReplacementRules, false);
     }
 
     /**
      * Creates a new InterceptorManager that replays test session records. It takes a set of
-     * {@code textReplacementRules}, that can be used by {@link PlaybackClient} to replace values in a
-     * {@link NetworkCallRecord#getResponse()}.
+     * {@code textReplacementRules}, that can be used by {@link PlaybackClient} to replace values in a {@link
+     * NetworkCallRecord#getResponse()}.
      *
      * The test session records are read from: "<i>session-records/{@code testName}.json</i>"
      *
      * @param testName Name of the test session record.
-     * @param textReplacementRules A set of rules to replace text in {@link NetworkCallRecord#getResponse()} when playing
-     * back network calls.
-     * @throws IOException An existing test session record could not be located or the data could not be deserialized
-     * into an instance of {@link RecordedData}.
+     * @param textReplacementRules A set of rules to replace text in {@link NetworkCallRecord#getResponse()} when
+     * playing back network calls.
+     * @param doNotRecord Flag indicating whether network calls should be record or played back.
+     * @throws UncheckedIOException An existing test session record could not be located or the data could not be
+     * deserialized into an instance of {@link RecordedData}.
      * @throws NullPointerException If {@code testName} or {@code textReplacementRules} is {@code null}.
      */
-    public InterceptorManager(String testName, Map<String, String> textReplacementRules) throws IOException {
```

Comment:
Removing `throws IOException` is a breaking change but this is core-test. I am not sure what is our policy for breaking changes in `core-test`.
cc @JonathanGiles

Reply:
Is removing a throw a breaking change? I agree changing it from throwing an IOException to UncheckedIOException is breaking.

Reply:
Yes, removing `throws CheckedException` from a method signature is a breaking change because you can `catch` a checked exception only if something in the `try` block throws it.